### PR TITLE
Adds a unit test that all roundstart spawnable jobs have a landmark to spawn at

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

https://github.com/tgstation/tgstation/pull/74985#issuecomment-1523773626

Tests that all jobs that have `spawn_position > 0` at roundstart have a location to spawn at. 

Also changes the failure message for `get_default_roundstart_spawn_point` to send to `log_mapping` rather than `log_world`, as it is a map error and log world doesn't really help anyone. This would've been sufficient for the existing unit test `log_mapping`, but that unit test expects that the log has an areacoord supplied, which we can't really do, given we're reporting a _lack_ of something. 

## Why It's Good For The Game

Stops maps from regressing and dumping people on the latejoin shuttle roundstart. 

## Changelog

~~Not necessary unless I find a map which forgot a landmark~~

:cl: Melbert
fix: Tramstation Robotics and RD now spawn in their departments roundstart
fix: Birdboat detective now spawns in their office
/:cl:

### CODE REVIEW

**Review by san7890 — APPROVED** [review]
lol

File: code/modules/unit_tests/mapping.dm [hunk 1]
```diff
@@ -18,3 +18,16 @@
 
 		TEST_FAIL(log_entry)
 
+
+/// Tests that [/datum/job/proc/get_default_roundstart_spawn_point] returns a landmark from all joinable jobs.
+/datum/unit_test/job_roundstart_spawnpoints
```
Comment (Mothblocks) [inline code/modules/unit_tests/mapping.dm:23]: Should make this its own file, I don't want this to be a catch all file for mapping problems

**Review by Mothblocks — COMMENTED** [review]
Are we sure that the debug maps should need to have every roundstart spawn point?? That seems wrong, we're making this worse. Debug maps should not have to worry about the results of this test (my best idea would be placing a marker datum that if it exists skips this test, which might be a nice thing to add as a generic API as well)

**Comment by Mothblocks** [comment]
E.g. `/obj/skip_test{test_name="job_roundstart_spawnpoints"}`

**Comment by Mothblocks** [comment]
Let's do that then, I think you should do that with your other PR too

**Pull request merged**